pkg/controller/azvolume.go

```diff
@@ -247,71 +246,67 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 		}
 		azVolume = updatedObj.(*azdiskv1beta2.AzVolume)
 
-		if volumeDeleteRequested {
-			w.Logger().V(5).Info("Deleting Volume...")
-		}
+		w.Logger().V(5).Info("Deleting Volume...")
+	}
 
-		waitCh := make(chan goSignal)
-		//nolint:contextcheck // call is asynchronous; context is not inherited by design
-		go func() {
-			_, goWorkflow := workflow.New(ctx)
-			var deleteErr error
-			defer func() { goWorkflow.Finish(deleteErr) }()
-			waitCh <- goSignal{}
+	waitCh := make(chan goSignal)
+	//nolint:contextcheck // call is asynchronous; context is not inherited by design
+	go func() {
+		_, goWorkflow := workflow.New(ctx)
 
-			goCtx := goWorkflow.SaveToContext(context.Background())
+		var deleteErr error
+		defer func() { goWorkflow.Finish(deleteErr) }()
+		waitCh <- goSignal{}
 
-			deleteCtx, deleteCancel := context.WithTimeout(goCtx, cloudTimeout)
-			defer deleteCancel()
+		goCtx := goWorkflow.SaveToContext(context.Background())
 
-			reportError := func(obj interface{}, err error) error {
-				azv := obj.(*azdiskv1beta2.AzVolume)
-				_ = r.updateError(azv, err)
-				_, derr := r.updateState(azv, azdiskv1beta2.VolumeDeletionFailed, forceUpdate)
-				return derr
-			}
+		var updateFunc azureutils.UpdateCRIFunc
+		var err error
+		updateMode := azureutils.UpdateCRI
 
-			var updateFunc azureutils.UpdateCRIFunc
-			var err error
-			updateMode := azureutils.UpdateCRIStatus
+		reportError := func(obj interface{}, err error) error {
```

Comment:
Updating AzVolume CRI with error with new error and then updating the state seems like a common operation. Why are we defining a new function for within a method rather than creating a separate function we can use more consistently.